src/Sylius/Component/Taxation/Resolver/TaxRateResolver.php

```diff
@@ -13,14 +13,18 @@
 
 namespace Sylius\Component\Taxation\Resolver;
 
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
 use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Taxation\Checker\TaxRateDateCheckerInterface;
 use Sylius\Component\Taxation\Model\TaxableInterface;
 use Sylius\Component\Taxation\Model\TaxRateInterface;
 
 class TaxRateResolver implements TaxRateResolverInterface
 {
-    public function __construct(protected RepositoryInterface $taxRateRepository)
-    {
+    public function __construct(
+        protected RepositoryInterface $taxRateRepository,
+        protected TaxRateDateCheckerInterface $taxRateDateChecker
```

Comment:
To provide a BC layer, we need to make it nullable and check in the `resolve` method. Also throw a deprecation when it is not passed